src/lambda/wizards/samInitWizard.ts

```diff
@@ -482,7 +483,9 @@ export class CreateNewSamAppWizard extends MultiStepWizard<CreateNewSamAppWizard
     }
 
     private readonly NAME: WizardStep = async () => {
-        this.name = await this.context.promptUserForName()
+        this.name = await this.context.promptUserForName(
+            this.name ?? (this.location ? path.basename(this.location.path) : '')
```

Comment:
Not huge on the location-as-default-name, you'll have a nested directory with the same name. Might be better to have a generic "newSamApp" or something as default text.

Reply:
> you'll have a nested directory with the same name

this is somewhat common though, for e.g. python and csharp ecosystems. Ask: which one is the user more likely to change? "new sam app" they *always* have to change, whereas location-as-default-name is a reasonable default.

do we want to force users to always change the name?